lib/Display/BitmapRegion.cpp

```diff
@@ -38,3 +60,48 @@ void BitmapRegion::render(GxEPD2_GFX* display) {
         backgroundColor);
   }
 }
+
+Rectangle BitmapRegion::getBoundingBox() {
+  return {
+    .x = std::min(this->boundingBox.x, this->previousBound.x),
+    .y = std::min(this->boundingBox.y, this->previousBound.y),
+    .w = std::max(this->boundingBox.w, this->previousBound.w),
+    .h = std::max(this->boundingBox.h, this->previousBound.h)
+  };
+}
+
+bool BitmapRegion::getBitmapDimensions(const String& bitmapName, uint16_t *width, uint16_t *height){
+  String metadata_filename = bitmapName.c_str();
+  Serial.println("AAAAAAAAAAAA");
+  
+  metadata_filename.replace("/b/", "/m/");
+  Serial.println(F("getting dimensions of bitmap:"));
+    Serial.println(bitmapName);
+    Serial.println(metadata_filename);
+  if (!SPIFFS.exists(bitmapName) || !SPIFFS.exists(metadata_filename)) {
+    Serial.println(F("WARN - tried to get dimensions from bitmap that doesn't exist: "));
+    Serial.println(bitmapName);
+    Serial.println(metadata_filename);
+    return false;
```

Comment:
Calculate default based on bitmap file size

Reply:
Should we strike this or keep it?

Reply:
I see three options:

1. Calculate the default somewhere (was thinking this was the appropriate place)
2. Calculate a default on the API side and generate a default metadata file so that we can treat the existence of a metadata file as an invariant (in which case what you have here is fine)
3. Refuse to save bitmaps that don't have metadata files supplied (same deal -- we get the invariant)

I was originally proposing (1), but (2) hadn't occurred to me until now, and I think I like that better.  Open to whatever makes most sense to you, though.

Reply:
~~2, I feel is best. I think it should save the bitmap even if there isn't a supplied W and H with a default set by us ~~

Actually, no. 3. Since if you use the resizer in the bitmap editor, it will screw up the already supplied bitmap. 

Thoughts?

Reply:
The UI [always submits the dimensions](https://github.com/sidoh/epaper_templates/blob/master/web/src/bitmaps/BitmapsIndex.jsx#L127-L152) of the image every time it's saved, so that shouldn't be a problem.

Either (2) or (3) seem fine.